src/py/kaleido/_fig_tools.py

```diff
@@ -13,6 +13,10 @@
 SUPPORTED_FORMATS = ("png", "jpg", "jpeg", "webp", "svg", "json", "pdf")  # pdf and eps
 
 
+def _is_figurish(o):
+    return hasattr(o, "to_dict") or (isinstance(o, dict) and "data" in o)
```

Comment:
@ayjayt A dict without a `data` key could still be a valid figure object (or at least, it can be passed to `Plotly.newPlot` without error).

If anything we should just check that each key in the dict is one of the 4 field names supported by `Plotly.newPlot` (see docs).

https://plotly.com/javascript/plotlyjs-function-reference/#plotlynewplot

```suggestion
    return hasattr(o, "to_dict") or (isinstance(o, dict) and k in {"data", "layout", "config", "frames"} for k in o.keys())
```

Or... is that validation even needed? What if you just pass along anything that's a `dict` and let whatever happens happen?
```suggestion
    return hasattr(o, "to_dict") or isinstance(o, dict)
```

Reply:
> Or... is that validation even needed? What if you just pass along anything that's a dict and let whatever happens happen?

User gets some intimidating error referencing internals and they make an issue lol

Reply:
fair enough! I've realized this line is fine as-is since the Kaleido js frontend requires both `data` and `layout` keys anyway (though given that, maybe we should check for both `data` and `layout` here as well?).

Reply:
Yeah that sounds good. I'm actually probably gonna get to it this weekend.